### PR TITLE
adds class tc-checkbox-checked when checked

### PR DESCRIPTION
**Pull request by tobibeer**
* equivalent to #2182 (RadioWidget)
* also applies `tc-checkbox` to checkboxes by default, always

### CODE REVIEW

File: core/modules/widgets/checkbox.js [hunk 1]
```diff
@@ -33,13 +33,15 @@ CheckboxWidget.prototype.render = function(parent,nextSibling) {
...
 	this.execute();
+	var isChecked = this.getValue();
 	// Create our elements
 	this.labelDomNode = this.document.createElement("label");
-	this.labelDomNode.setAttribute("class",this.checkboxClass);
+	this.labelDomNode.setAttribute("class","tc-checkbox " + this.checkboxClass);
 	this.inputDomNode = this.document.createElement("input");
 	this.inputDomNode.setAttribute("type","checkbox");
-	if(this.getValue()) {
+	if(isChecked) {
```
Comment (ght) [inline core/modules/widgets/checkbox.js:42]: This is the only use in this scope, so it should not be lifted out into a var.

File: editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid [hunk 2]
```diff
@@ -25,7 +25,7 @@ The content of the `<$checkbox>` widget is displayed within an HTML `<label>` el
 |checked |The value of the field corresponding to the checkbox being checked |
 |unchecked |The value of the field corresponding to the checkbox being unchecked |
 |default |The default value to use if the field is not defined |
-|class |The class that will be assigned to the label element |
+|class |The class that will be assigned to the label element <$macrocall $name=".tip" _="""<<.from-version "5.1.14">> By default, always set to `tc-checkbox`, as well as `tc-checkbox-checked` when checked"""/>|
```
Comment (ght) [inline editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid:28]: Version needs to be updated

**Review by ght — COMMENTED** [review]
Not sure if this change is still wanted, since the PR was dormant for so long.

But if it is, it's a small change and looks like it can be merged with only minor adjustments.

**Comment by Jermolene** [comment]
Thanks @ght 

@rmunn is working on some enhancements to the checkbox widget, perhaps it might be worth integrating this functionality so that you don't need to rebase?

**Comment by rmunn** [comment]
Sure, I can integrate this change into #6561 if you want.

**Comment by Jermolene** [comment]
Come to think of it, the original reason for not merging this was probably that it is already possible to distinguish checked/unchecked checkboxes in CSS selectors using the `:checked` and `:unchecked` pseudo-classes. So perhaps on reflections lets not bother, unless somebody can propose a usecase that isn't currently possible.

**Comment by rmunn** [comment]
Does that CSS selector apply to the label as well? I could see someone wanting to make the label bold when the checkbox is checked.

... I see https://developer.mozilla.org/en-US/docs/Web/CSS/:checked has a `:checked + label` syntax. I'll have to experiment.

**Comment by Jermolene** [comment]
> Does that CSS selector apply to the label as well?

Ah yes of course, you're right. This PR adds the class to the label, not the checkbox. So I agree it does make sense.

**Comment by rmunn** [comment]
The code from this PR is now in #6561. I forgot to do anything special with `git commit --authot`, so the commit that implements this is currently by Robin Munn. So I'm about to amend that commit and force-push the branch from #6561 so that credit for this commit goes to @tobibeer. Then #6561 will be ready for review.

**Comment by rmunn** [comment]
The code from this PR is now in #6561. I forgot to do anything special with `git commit --authot`, so the commit that implements this is currently by Robin Munn. So I'm about to amend that commit and force-push the branch from #6561 so that credit for this commit goes to @tobibeer. Then #6561 will be ready for review.

**Comment by pmario** [comment]
@Jermolene ... This PR is outdated and the latest "rebuild" of the checkbox widget takes care of this class setting.

**Comment by Jermolene** [comment]
Thanks @pmario

**Pull request closed without merging**